### PR TITLE
Add security policy

### LINKED ISSUE #311: Security Address
Hello!

I may have found a security issue in latest version of [pyogrio](https://pypi.org/project/pyogrio). Following responsible disclosure, is there an email or other private channel where I could share the details?
Thank you

### PR DESCRIPTION
**Pull request by brendan-ward**
Resolves #311

Loosely inspired by [Fiona #1308](https://github.com/Toblerity/Fiona/pull/1308).

This is perhaps a bit on the verbose side, but I was trying to provide sufficient content in regards to 3rd party dependencies, since that is where the majority of the vulnerabilities are likely to occur.

Notably absent is what our follow-up timeframe is, since I didn't want to obligate us to follow a specific response protocol or timeframe.  Rather, we can refine that protocol on a case by case basis depending on the nature of vulnerabilities as they are reported.

### CODE REVIEW

**Comment by brendan-ward** [comment]
@SCH227 does this policy seem sufficient for now?

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+When you build Pyogrio locally, you link it to your local version of GDAL and
+associated dependencies in a way that depends on how you installed GDAL and its
+dependencies (e.g., system package, homebrew package, etc). Please contact the
+appropriate maintainers of these packages to report vulnerabilities.
+
+## Supported versions
+
+Pyogrio has not yet reached 1.0. Only the latest available version is being
+supported with security updates.
```
Comment (martinfleis) [inline SECURITY.md:50]: Will this ever be different? I.e. do you assume we will support older major versions by patching them?

Comment (brendan-ward) [reply SECURITY.md:50]: Once we hit 1.0, we _may_ consider backporting fixes to a previous (but still relatively recent) version.

**Review by martinfleis — APPROVED** [review]
Thanks!

**Comment by SCH227** [comment]
@brendan-ward it looks greats to me, thank you!

**Review by theroggy — APPROVED** [review]

**Pull request merged**